source/common/router/config_impl.h

```diff
@@ -547,6 +547,27 @@ class InternalRedirectPolicyImpl : public InternalRedirectPolicy {
 };
 using DefaultInternalRedirectPolicy = ConstSingleton<InternalRedirectPolicyImpl>;
 
+/**
+ * Container for route config elements that pertain to a redirect.
+ *
+ * We keep them in a separate data structure to avoid memory overhead for the routes that do not use
+ * redirect.
+ */
+struct RedirectConfig {
+  RedirectConfig(const envoy::config::route::v3::Route& route);
+  const std::string scheme_redirect_;
+  const std::string host_redirect_;
+  const std::string port_redirect_;
+  const std::string path_redirect_;
+  const std::string prefix_rewrite_redirect_;
+  std::string regex_rewrite_redirect_substitution_;
```

Comment:
These 3 are defined as `oneof`, so further optimization may be possible here, but maybe as a future work.

Reply:
Future sgtm.